tensorflow/core/kernels/data/range_dataset_op_test.cc

```diff
@@ -38,9 +39,26 @@ class RangeDatasetOpTest : public DatasetOpsTestBase {
 };
 
 struct TestCase {
```

Comment:
I think it would be more modular to define different TestCase structs for different types of tests. For instance `IsStateful` can only ever return true. If you would like to avoid duplication, you can create a base test case (which defines `start`, `stop`, and `step`) and test cases for different APIs derive from it if necessary. 

That way, when a new API is introduced (similar to `IsStateful`), we do not have to update all existing test cases and the test case constructor and we instead create a new test case class derived from the base test case class.

Reply:
It is a good idea to avoid the updating of all existing test cases when a new API is introduced. 

IIUC, there may be an issue If we define different TestCase structs for different types of tests: we will have 12 TestCase structs; For each API test, we need to create at least one TestCase instance and these TestCase instances have some duplications in the dataset input parameters (e.g. start, stop, step). That means we will have more than 12 test cases for each dataset op test if different scenarios need to be tested.

Reply:
That's why I was suggesting to create a `BaseTestCase` base class which has the parameters shared by all tests and derive from it to create test cases for specific API endpoints.

Reply:
I will submit a commit for the modular TestCase structs. We can discuss it based on the code.

Reply:
@jsimsa The commit (https://github.com/tensorflow/tensorflow/commit/01830d8bec1ab7e00214bc9eafcc37fe50b0475b) is submitted. Please feel free to review and comment!